### PR TITLE
feat(numpy backend): removed manual dtype casting.

### PR DESCRIPTION
**Pull request by Madjid-CH**
# PR Description
used script for dtypes checking:
``` 
import numpy as np

dtypes = [
    np.dtype("int8"),
    np.dtype("int16"),
    np.dtype("int32"),
    np.dtype("int64"),
    np.dtype("uint8"),
    np.dtype("uint16"),
    np.dtype("uint32"),
    np.dtype("uint64"),
    np.dtype("float16"),
    np.dtype("float32"),
    np.dtype("float64"),
    np.dtype("complex64"),
    np.dtype("complex128"),
    np.dtype("bool"),
]


def get_supported_and_unsupported_dtypes():
    supported_dtypes = set()
    unsupported_dtypes = set()

    for dtype in dtypes:
        try:
            x = np.array(
                [[1, 2], [2, 1]],
                dtype=dtype,
            )
            x1 = np.array([1, 1, 1], dtype=dtype)
            x2 = np.array([1, 2, 3], dtype=dtype)
            print(dtype,  cummax(x1))
            supported_dtypes.add(dtype)
        except Exception as e:
            print(e)
            unsupported_dtypes.add(dtype)

    return supported_dtypes, unsupported_dtypes


if __name__ == "__main__":
    supported_dtypes, unsupported_dtypes = get_supported_and_unsupported_dtypes()

    supported_dtypes = sorted(str(d) for d in supported_dtypes)
    unsupported_dtypes = sorted(str(d) for d in unsupported_dtypes)


    print("Supported Data Types:")
    m = f"""@with_supported_dtypes(
            {{"1.25.2 and below": {tuple(supported_dtypes)}}}, backend_version
        )"""
    print(m)

    print("\nUnsupported Data Types:")
    m = f"""@with_unsupported_dtypes(
            {{"1.25.2 and below": {tuple(unsupported_dtypes)}}}, backend_version
        )"""
    print(m)

```

### CODE REVIEW

File: ivy/functional/backends/numpy/experimental/statistical.py [hunk 1]
```diff
@@ -428,13 +428,6 @@ def cummax(
     dtype: Optional[np.dtype] = None,
     out: Optional[np.ndarray] = None,
 ) -> Tuple[np.ndarray, np.ndarray]:
```
Comment (Madjid-CH) [inline ivy/functional/backends/numpy/experimental/statistical.py:430]: result of running the script on `cummax`:
```
int8 (array([1, 1, 1], dtype=int8), array([0, 1, 2], dtype=int64))
int16 (array([1, 1, 1], dtype=int16), array([0, 1, 2], dtype=int64))
int32 (array([1, 1, 1]), array([0, 1, 2], dtype=int64))
int64 (array([1, 1, 1], dtype=int64), array([0, 1, 2], dtype=int64))
uint8 (array([1, 1, 1], dtype=uint8), array([0, 1, 2], dtype=int64))
uint16 (array([1, 1, 1], dtype=uint16), array([0, 1, 2], dtype=int64))
uint32 (array([1, 1, 1], dtype=uint32), array([0, 1, 2], dtype=int64))
uint64 (array([1, 1, 1], dtype=uint64), array([0, 1, 2], dtype=int64))
float16 (array([1., 1., 1.], dtype=float16), array([0, 1, 2], dtype=int64))
float32 (array([1., 1., 1.], dtype=float32), array([0, 1, 2], dtype=int64))
float64 (array([1., 1., 1.]), array([0, 1, 2], dtype=int64))
complex64 (array([1.+0.j, 1.+0.j, 1.+0.j], dtype=complex64), array([0, 1, 2], dtype=int64))
complex128 (array([1.+0.j, 1.+0.j, 1.+0.j]), array([0, 1, 2], dtype=int64))
bool (array([ True,  True,  True]), array([0, 1, 2], dtype=int64))
Supported Data Types:
@with_supported_dtypes(
        {"1.25.2 and below": ('bool', 'complex128', 'complex64', 'float16', 'float32', 'float64', 'int16', 'int32', 'int64', 'int8', 'uint16', 'uint32', 'uint64', 'uint8')}, backend_version
    )
Unsupported Data Types:
@with_unsupported_dtypes(
        {"1.25.2 and below": ()}, backend_version
    )

```

File: ivy/functional/backends/numpy/experimental/statistical.py [hunk 2]
```diff
@@ -527,10 +520,7 @@ def cummin(
     out: Optional[np.ndarray] = None,
 ) -> np.ndarray:
     if dtype is None:
```
Comment (Madjid-CH) [inline ivy/functional/backends/numpy/experimental/statistical.py:522]: same for `cummin`
```
int8 [1 1 1]
int16 [1 1 1]
int32 [1 1 1]
int64 [1 1 1]
uint8 [1 1 1]
uint16 [1 1 1]
uint32 [1 1 1]
uint64 [1 1 1]
float16 [1. 1. 1.]
float32 [1. 1. 1.]
float64 [1. 1. 1.]
complex64 [1.+0.j 1.+0.j 1.+0.j]
complex128 [1.+0.j 1.+0.j 1.+0.j]
bool [ True  True  True]
Supported Data Types:
@with_supported_dtypes(
        {"1.25.2 and below": ('bool', 'complex128', 'complex64', 'float16', 'float32', 'float64', 'int16', 'int32', 'int64', 'int8', 'uint16', 'uint32', 'uint64', 'uint8')}, backend_version
    )
Unsupported Data Types:
@with_unsupported_dtypes(
        {"1.25.2 and below": ()}, backend_version
    )
```

File: ivy/functional/backends/numpy/statistical.py [hunk 3]
```diff
@@ -171,7 +171,7 @@ def var(
 # ------#
 
 
-@with_unsupported_dtypes({"1.25.2 and below": "bfloat16"}, backend_version)
```
Comment (Madjid-CH) [inline ivy/functional/backends/numpy/statistical.py:174]:
```
int8 [1 2 6]
int16 [1 2 6]
int32 [1 2 6]
int64 [1 2 6]
uint8 [1 2 6]
uint16 [1 2 6]
uint32 [1 2 6]
uint64 [1 2 6]
float16 [1. 2. 6.]
float32 [1. 2. 6.]
float64 [1. 2. 6.]
complex64 [1.+0.j 2.+0.j 6.+0.j]
complex128 [1.+0.j 2.+0.j 6.+0.j]
bool [ True  True  True]
Supported Data Types:
@with_supported_dtypes(
        {"1.25.2 and below": ('bool', 'complex128', 'complex64', 'float16', 'float32', 'float64', 'int16', 'int32', 'int64', 'int8', 'uint16', 'uint32', 'uint64', 'uint8')}, backend_version
    )
Unsupported Data Types:
@with_unsupported_dtypes(
        {"1.25.2 and below": ()}, backend_version
    )
```

File: ivy/functional/backends/numpy/statistical.py [hunk 4]
```diff
@@ -218,10 +215,6 @@ def cumsum(
     out: Optional[np.ndarray] = None,
 ) -> np.ndarray:
     if dtype is None:
-        if x.dtype == "bool":
```
Comment (Madjid-CH) [inline ivy/functional/backends/numpy/statistical.py:221]:
``` 
int8 [1 3 6]
int16 [1 3 6]
int32 [1 3 6]
int64 [1 3 6]
uint8 [1 3 6]
uint16 [1 3 6]
uint32 [1 3 6]
uint64 [1 3 6]
float16 [1. 3. 6.]
float32 [1. 3. 6.]
float64 [1. 3. 6.]
complex64 [1.+0.j 3.+0.j 6.+0.j]
complex128 [1.+0.j 3.+0.j 6.+0.j]
bool [ True  True  True]
Supported Data Types:
@with_supported_dtypes(
        {"1.25.2 and below": ('bool', 'complex128', 'complex64', 'float16', 'float32', 'float64', 'int16', 'int32', 'int64', 'int8', 'uint16', 'uint32', 'uint64', 'uint8')}, backend_version
    )
Unsupported Data Types:
@with_unsupported_dtypes(
        {"1.25.2 and below": ()}, backend_version
    )
```

**Review by vedpatwardhan — APPROVED** [review]
lgtm! Feel free to merge, thanks @Madjid-CH 😄
(I'm assuming you've already run the tests for the functions changed and the tests pass or fail, whichever was the state of those tests before the changes. And even if it fails, the logs of the failure are the same as those before making the changes.)

**Comment by Madjid-CH** [comment]
> lgtm! Feel free to merge, thanks @Madjid-CH 😄 (I'm assuming you've already run the tests for the functions changed and the tests pass or fail, whichever was the state of those tests before the changes. And even if it fails, the logs of the failure are the same as those before making the changes.)

yeah, the tests was failing because of a missing `on_device ` attribute and now they are passing except for `cummax`, it test is still failing after modification.

**Pull request merged**